### DISCUSSION TITLE
How does `ClientServer` mode gets Vulnerability DB updates?

### DISCUSSION THREAD

**Discussion by sheeeng** [aquasecurity/trivy-operator#2096] [Help & Support] [answered]
How does `ClientServer` mode gets Vulnerability DB updates?

**Comment by chen-keinan** [comment]
@sheeeng the trivy server take care of updates automatically

**Reply by sheeeng** [reply]
What is the interval for updates?

**Reply by sheeeng** [reply]
How could we verify that the trivy server is using updated Vulnerability DB?

**Reply by chen-keinan** [reply]
6h

**Reply by chen-keinan** [reply]
there should be `metadata.json` under cache folder look like this:
```json
{
  "Version": 2,
  "NextUpdate": "2024-05-22T06:18:28.021341728Z",
  "UpdatedAt": "2024-05-22T00:18:28.021342039Z",
  "DownloadedAt": "2024-05-22T05:46:03.191826Z"
}
```

**Reply by sheeeng** [reply]
Thank you for the information, @chen-keinan.

Where is the cache folder located by default?

**Reply by chen-keinan** [reply]
are you running built-in server in trivy-operator ?  or running your own trivy server ?

**Reply by sheeeng** [reply]
> are you running built-in server in trivy-operator ? or running your own trivy server ?

The built-in server in trivy-operator.

I don't see any default path specified. https://github.com/aquasecurity/trivy-operator/blob/a8b5e42aec1e699b89713a519dc3224a853507da/deploy/helm/values.yaml#L569-L593

**Reply by chen-keinan** [reply]
cache should be [here](https://github.com/aquasecurity/trivy-operator/blob/main/deploy/helm/templates/trivy-server/statefulset.yaml#L115):

**Reply by sheeeng** [reply, marked as answer]
Thank you again for the tip, @chen-keinan. I've located the `metadata.json` file.

https://github.com/aquasecurity/trivy-operator/blob/a8b5e42aec1e699b89713a519dc3224a853507da/deploy/helm/templates/trivy-server/statefulset.yaml#L115

```console
$ kubectl exec --namespace trivy-system --stdin --tty pod/trivy-server-0 --container trivy-server -- /bin/sh
$
$ cat /home/scanner/.cache/trivy/db/metadata.json
{"Version":2,"NextUpdate":"2024-05-23T06:16:13.858957998Z","UpdatedAt":"2024-05-23T00:16:13.858958288Z","DownloadedAt":"2024-05-23T00:40:46.462412524Z"}
```
